packages/config/src/Config.ts

```diff
@@ -381,51 +385,46 @@ const APP_JSON_EXAMPLE = JSON.stringify({
 
 function ensureConfigHasDefaultValues(
   projectRoot: string,
-  exp: ExpoConfig,
+  exp: Partial<ExpoConfig>,
   pkg: JSONObject,
   skipSDKVersionRequirement: boolean = false
 ): { exp: ExpoConfig; pkg: PackageJSONConfig } {
-  if (!exp) exp = {};
-
-  if (!exp.name) {
-    if (typeof pkg.name !== 'string') {
-      pkg.name = path.basename(projectRoot);
-    }
-    exp.name = pkg.name;
+  // Defaults for package.json fields
+  const pkgName = typeof pkg.name === 'string' ? pkg.name : path.basename(projectRoot);
+  const pkgVersion = typeof pkg.version === 'string' ? pkg.version : '1.0.0';
+
+  const pkgWithDefaults = { ...pkg, name: pkgName, version: pkgVersion };
+
+  // Defaults for app.json/app.config.js fields
+  const name = exp.name ?? pkgName;
+  const slug = exp.slug ?? slugify(name.toLowerCase());
+  const version = exp.version ?? pkgVersion;
+  const nodeModulesPath = exp.nodeModulesPath
+    ? path.resolve(projectRoot, exp.nodeModulesPath)
+    : undefined;
+  let description = exp.description;
+  if (!description && typeof pkg.description === 'string') {
+    description = pkg.description;
   }
 
-  if (!exp.description && typeof pkg.description === 'string') {
-    exp.description = pkg.description;
-  }
-
-  if (!exp.slug && typeof exp.name === 'string') {
-    exp.slug = slug(exp.name.toLowerCase());
-  }
-
-  if (!exp.version) {
-    if (typeof pkg.version === 'string') {
-      exp.version = pkg.version;
-    } else {
-      pkg.version = '1.0.0';
-    }
-    exp.version = pkg.version;
-  }
-
-  if (exp.nodeModulesPath) {
-    exp.nodeModulesPath = path.resolve(projectRoot, exp.nodeModulesPath);
-  }
```

Comment:
Same question as above - this function was intentionally or unintentionally modifying the `exp` object.

Reply:
I think it was unintentional and it's usually better to not modify objects passed in a arguments to avoid problems.